rules/static.go

```diff
@@ -411,23 +449,29 @@ func (r *DefaultRuleRenderer) filterOutputChain(ipVersion uint8) *Chain {
 	rules = append(rules, r.acceptAlreadyAccepted()...)
 
 	if r.KubeIPVSSupportEnabled {
-		// Jump to from-endpoint-mark dispatch chain if endpoint mark is not zero, which means
+		if ipVersion == 4 && r.IPIPEnabled {
+			// Auto-allow IPIP traffic from tunnel device to other host interface.
+			// If IPIP is enabled, the packet will be sent over to tunnel device and
+			// come back with encapsulated format ( include mark bits been copied over) through OUTPUT filter chain.
+			// We need to make sure packet been allowed to stop it going through from-endpoint-mark again with node ip.
+			rules = append(rules,
+				Rule{
+					Match: Match().ProtocolNum(ProtoIPIP).
+						SourceIPSet(r.IPSetConfigV4.NameForMainIPSet(IPSetIDThisHostIPs)).
+						DestIPSet(r.IPSetConfigV4.NameForMainIPSet(IPSetIDAllHostIPs)).
+						SrcAddrType(AddrTypeLocal, false),
+					Action:  r.filterAllowAction,
+					Comment: "Allow IPIP packets from tunnel device to other Calico hosts",
+				},
+			)
+		}
+
+		// Jump to forward-endpoint-mark chain if endpoint mark is not zero, which means
 		// packet has been through filter INPUT chain. There could be policies apply to its ingress interface.
 		rules = append(rules,
 			Rule{
 				Match:  Match().MarkNotClear(r.IptablesMarkEndpoint),
```

Comment:
I think we might need to clear the mark in case the user has their own tunnels set up.  For example, what if they're using a GRE tunnel and the GRE packets go through the OUTPUT chain again, just like our IPIP packets?

Reply:
Done. Replace `ReturnAction`  to `GotoAction{Target: cali-clear-endpoint-mark}` .